src/main/java/portablejim/bbw/core/OopsCommand.java

```diff
@@ -39,50 +40,28 @@ public void processCommand(ICommandSender sender, String[] arguments) {
             if (currentItemstack != null && currentItemstack.getItem() != null
                     && currentItemstack.getItem() instanceof IWandItem) {
                 NBTTagCompound tagComponent = currentItemstack.getTagCompound();
-
-                NBTTagCompound bbwCompound;
                 if (tagComponent != null && tagComponent.hasKey("bbw", Constants.NBT.TAG_COMPOUND)
                         && tagComponent.getCompoundTag("bbw").hasKey("lastPlaced", Constants.NBT.TAG_INT_ARRAY)) {
-                    bbwCompound = tagComponent.getCompoundTag("bbw");
-                    ArrayList<Point3d> pointList = unpackNbt(bbwCompound.getIntArray("lastPlaced"));
-                    for (Point3d point : pointList) {
-                        player.getEntityWorld().setBlockToAir(point.x, point.y, point.z);
-                    }
+                    NBTTagCompound bbwCompound = tagComponent.getCompoundTag("bbw");
                     if (bbwCompound.hasKey("lastBlock", Constants.NBT.TAG_STRING)
                             && bbwCompound.hasKey("lastPerBlock", Constants.NBT.TAG_INT)) {
                         GameRegistry.UniqueIdentifier lastBlock = new GameRegistry.UniqueIdentifier(
                                 bbwCompound.getString("lastBlock"));
                         int damageValue = bbwCompound.getInteger("lastDamage");
                         ItemStack itemStack = GameRegistry.findItemStack(lastBlock.modId, lastBlock.name, 1);
                         itemStack.setItemDamage(damageValue);
-                        int count = bbwCompound.getInteger("lastPerBlock") * pointList.size();
-                        int stackSize = itemStack.getMaxStackSize();
-                        int fullStacks = count / stackSize;
-                        for (int i = 0; i < fullStacks; i++) {
-                            ItemStack newStack = itemStack.copy();
-                            newStack.stackSize = stackSize;
-                            player.worldObj.spawnEntityInWorld(
-                                    new EntityItem(
-                                            player.getEntityWorld(),
-                                            player.posX,
-                                            player.posY,
-                                            player.posZ,
-                                            newStack));
+                        Block block = Block.getBlockFromItem(itemStack.getItem());
+                        World world = player.worldObj;
+                        ArrayList<Point3d> pointList = unpackNbt(bbwCompound.getIntArray("lastPlaced"));
+                        int count = 0;
+                        for (Point3d point : pointList) {
+                            if (world.getBlock(point.x, point.y, point.z) == block) {
```

Comment:
The Block class does not compare metadata and damage.
This will allow player to cheese the same blocks which have different metadata.


https://github.com/GTNewHorizons/BetterBuildersWands/assets/39122497/2730d4ab-591c-4edc-97e2-f9859bb713a7

Reply:
It should be good now